media/js/webauthn-mfa.js

```diff
@@ -201,12 +201,9 @@ akeeba.MFA.webauthn.onValidateClick = function (event)
     return false;
 }
 
-document.getElementById("passkeys-missing").style.display = "none";
-
-if (typeof (navigator.credentials) == "undefined")
-{
-    document.getElementById("passkeys-missing").style.display  = "block";
-    document.getElementById("passkeys-controls").style.display = "none";
+if (typeof (navigator.credentials) == "undefined") {
```

Comment:
The line you removed needs to be put back.

The missing passkeys message is visible **by default**. This way, if the Javascript fails to load, or does not run for any reason the user sees a message that passkeys – which rely on JS – are not supported. When the JS loads we need to hide this message. If the `navigator.credentials` object is undefined, then the browser does load our JS but has no passkey support, therefore we display it again.

Another way to write that would be:

```js
if (typeof (navigator.credentials) == "undefined")
{
    document.getElementById("passkeys-missing").style.display  = "block";
    document.getElementById("passkeys-controls").style.display = "none";
    document.getElementById("passkeys_button").style.disabled = "disabled";
}
else
{
    document.getElementById("passkeys-missing").style.display = "none";
}
```

However, it's best practice to avoid `else` blocks. Watch Rafael's presentation on code calisthentics from J and Beyond about a decade ago, he explained it far better than I can.

Reply:
i have put the js back that makes it visible by default and set to d-none whn the js runs